coralproject/talk
# [CORL-3061] Allow external mod phases to set `moderationAction`'s and interoperate with DSA mode active

## What does this PR do?

- Allow external mod phases to run when DSA is enabled
- Allow external mod phases to set the `moderationAction` which will create moderation actions when rejecting/approving comments
- Allow users to set the `rejectionReason` when supplying a `moderationAction`

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

None

## How do I test this PR?

- Spin up `utilities/externalModPhase` with an `npm i && npm run start` in that directory
- Spin up Coral
- Go to the `Admin` and enable DSA under `General > Digital Services Act feature set`
- Under `Admin > Moderation Phases` Add an external mod phase pointing to `http://localhost:7000/api/rejectWithReason`
- Using a commenter account, post a comment
- See the comment get rejected
- Check the user's notifications and see that a notification is shown with how it was rejected

- Head back to admin and disable the external mod phase
- Make a new mod phase, point it to `http://localhost:7000/api/approve`
- Post a comment with the commenter again
- See that the comment is approved

## Were any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
